modules/core/src/main/java/io/fluo/core/log/TracingTransaction.java

```diff
@@ -0,0 +1,164 @@
+    this.clazz = clazz;
+    
+    log.trace("txid: {} begin() thread: {}", txid, Thread.currentThread().getId());
+
+    if(triggerRow != null){
+      log.trace("txid: {} trigger: {} {}", txid, triggerRow, triggerColumn);
+    }
+    
+    if(clazz != null){
+      log.trace("txid: {} class: {}", txid, clazz.getName());
+    }
+    
+  }
+
+  @Override
+  public Bytes get(Bytes row, Column column) {
+    Bytes ret = tx.get(row, column);
+    log.trace("txid: {} get({}, {}) -> {}", txid, row, column, ret);
+    return ret;
+  }
+
+  @Override
+  public Map<Column,Bytes> get(Bytes row, Set<Column> columns) {
+    Map<Column,Bytes> ret = tx.get(row, columns);
+    log.trace("txid: {} get({}, {}) -> {}", txid, row, columns, ret);
+    return ret;
+  }
+
+  @Override
+  public Map<Bytes,Map<Column,Bytes>> get(Collection<Bytes> rows, Set<Column> columns) {
+    Map<Bytes,Map<Column,Bytes>> ret = tx.get(rows, columns);
+    log.trace("txid: {} get({}, {}) -> {}", txid, rows, columns, ret);
+    return ret;
+  }
+
+  @Override
+  public RowIterator get(ScannerConfiguration config) {
+    // TODO log something better
+    log.trace("txid: {} get(ScannerConfiguration)", txid);
+    return tx.get(config);
+  }
+
+  @Override
+  public void setWeakNotification(Bytes row, Column col) {
+    log.trace("txid: {} setWeakNotification({}, {})", txid, row, col);
+    tx.setWeakNotification(row, col);
+  }
+
+  @Override
+  public void set(Bytes row, Column col, Bytes value) throws AlreadySetException {
+    log.trace("txid: {} set({}, {}, {})", txid, row, col, value);
+    tx.set(row, col, value);
+  }
+
+  @Override
+  public void delete(Bytes row, Column col) throws AlreadySetException {
+    log.trace("txid: {} delete({}, {})", txid, row, col);
+    tx.delete(row, col);
+  }
+
+  @Override
+  public void commit() throws CommitException {
+    try {
+      tx.commit();
+      committed = true;
+      log.trace("txid: {} commit() -> SUCCESSFUL commitTs: {}", txid, tx.getStats().getCommitTs());
+    } catch (CommitException ce) {
+      log.trace("txid: {} commit() -> UNSUCCESSFUL commitTs: {}", txid, tx.getStats().getCommitTs());
+
+      if (!log.isTraceEnabled() && triggerRow != null) {
+        collisionLog.trace("txid: {} trigger: {} {}", txid, triggerRow, triggerColumn);
+      }
+
+      if (!log.isTraceEnabled() && clazz != null) {
+        collisionLog.trace("txid: {} class: {}", txid, clazz.getName());
+      }
+
+      collisionLog.trace("txid: {} collisions: {}", txid, tx.getStats().getRejected());
+
+      throw ce;
+    }
+  }
+
+  @Override
+  public void close() {
+    log.trace("txid: {} close()", txid);
+    if (summaryLog.isTraceEnabled()) {
+      TxStats stats = tx.getStats();
+      String className = "N/A";
+      if (clazz != null)
+        className = clazz.getSimpleName();
+      summaryLog.trace("txid: {} thread : {} time: {} #read:TODO #ret: {} #set: {} #collisions: {} waitTime: {} %s committed: {} class: {}", txid, Thread
```

Comment:
This `TODO` in the logs is a little confusing.  How about putting something in the code instead and creating an issue?